tests/functional/test_linux_appimage.py

```diff
@@ -21,25 +21,19 @@
 import subprocess
 import pytest
 
-from PyInstaller.compat import is_py36
-
 
 @pytest.mark.linux
 @pytest.mark.parametrize('arch', ['x86_64'])
 def test_appimage_loading(tmp_path, pyi_builder_spec, arch):
     # Skip the test if appimagetool is not found
     appimagetool = pathlib.Path.home() / ('appimagetool-%s.AppImage' % arch)
-    if not is_py36:
-        appimagetool = str(appimagetool)
-    if not os.path.isfile(appimagetool):
+    if appimagetool.is_file():
         pytest.skip('%s not found' % appimagetool)
 
     # Ensure appimagetool is executable
-    if not os.access(appimagetool, os.X_OK):
-        st = os.stat(appimagetool)
-        os.chmod(appimagetool, st.st_mode | stat.S_IXUSR)
-
-    tmp_path = str(tmp_path)  # Fix for Python 3.5
+    if not appimagetool.touch():
```

Comment:
It [seems](https://github.com/python/cpython/blob/79d2e62c008446fbbc6f264bb8a30e2d38b6ff58/Lib/pathlib.py#L1289-L1308) this will work only if you pass `exist_ok=False`. But I am not sure of that change, is it a better way to check for execution rights?

Reply:
Hmm, I'm not sure now. I'll update it to `os.access(appimagetool)`.